app/services/brapi/v1/germplasm_queries.rb

```diff
@@ -122,17 +139,23 @@ def execute_statement(sql, atts)
     # at least until the previous one has been deallocated
     Thread.exclusive do
       @connection.prepare('brapi_statement', sql)
-      
-      if( (atts != nil) && !(atts.empty?) )
-        results = @connection.exec_prepared("brapi_statement", atts)
-      else
-        results = @connection.exec_prepared("brapi_statement")
+      begin
+        if( (atts != nil) && !(atts.empty?) )
+          results = @connection.exec_prepared("brapi_statement", atts)
+        else
+          results = @connection.exec_prepared("brapi_statement")
+        end
+      rescue PG::Error => e
+        @connection.exec("ROLLBACK") 
+        results = nil
+        Rails.logger.warn { "Encountered an error executing a BrAPI germplasm-related query: #{e.message} #{e.backtrace.join("\n")}" }
```

Comment:
Personally, I would rather put the exception class into `app/services/brapi` even though it technically is not a service. We did not use custom exceptions before and I don't think it makes sense to add new directory hierarchy just for this on class.

Another minor thing - it would probably be OK to put `rescue_from` invocations in `Brapi::BaseController` instead of specific controllers.